modules/entity/src/sorted_state_adapter.ts

```diff
@@ -72,8 +73,25 @@ export function createSortedStateAdapter<T>(selectId: any, sort: any): any {
   }
 
   function updateManyMutably(updates: Update<T>[], state: R): DidMutate;
-  function updateManyMutably(updates: any[], state: any): DidMutate {
+  function updateManyMutably(map: UpdateMap<T>, state: R): DidMutate;
+  function updateManyMutably(updatesOrMap: any[] | any, state: any): DidMutate {
     const models: T[] = [];
+    const updates: Update<T>[] =
+      updatesOrMap instanceof Array
+        ? updatesOrMap
+        : state.ids.reduce((changes: Update<T>[], id: string | number) => {
+            const change = updatesOrMap(state.entities[id]);
+            if (change && Object.keys(change).length > 0) {
+              return [
```

Comment:
Recommend changing this line to:
```ts
changes.push({ id, changes: change });

return changes;
```

That way we don't allocate and deallocate an array for every item we are upserting.

Reply:
Done, thanks for the feedback/reviews!

Reply:
@tdeschryver Thanks for all of the hard work! We really appreciate it! 😄